core/src/com/unciv/logic/map/mapgenerator/RiverGenerator.kt

```diff
@@ -16,17 +16,21 @@ class RiverGenerator(
     }
 
     fun spawnRivers() {
+        if (tileMap.values.none { it.isWater }) return
         val numberOfRivers = tileMap.values.count { it.isLand } / MAP_TILES_PER_RIVER
 
         var optionalTiles = tileMap.values.asSequence()
-                .filter { it.baseTerrain == Constants.mountain && it.isFarEnoughFromWater() }.toMutableList()
+            .filter { it.baseTerrain == Constants.mountain && it.isFarEnoughFromWater() }
```

Comment:
Why not use TileInfo.height?

Reply:
Good question, but not part of this pr :)